unittests/integration/builder_tester.cpp

```diff
@@ -132,6 +132,161 @@ CUDAQ_TEST(BuilderTester, checkSimple) {
     EXPECT_TRUE(counts.begin()->first == "101");
   }
 
+  // Check the built-in adjoint S-gate (sdg) on a qubit
+  // that begins in the 0-state.
+  {
+    auto builder = cudaq::make_kernel();
+    auto q = builder.qalloc();
+    // Place qubit in superposition state.
+    builder.h(q);
+    // Rotate around Z by -pi/2, twice. Total rotation of -pi.
+    builder.sdg(q);
```

Comment:
I don't think we need any updates to the C++ kernel_builder for sdg or tdg. In C++ users can just use s<adj> and t<adj>. 

So then I think we really only need updates at the pybind level.